src/mapping/tests/AxialGeoMultiscaleMappingTest.cpp

```diff
@@ -0,0 +1,130 @@
+#include <Eigen/Core>
+#include <algorithm>
+#include <memory>
+#include "logging/LogMacros.hpp"
+#include "mapping/AxialGeoMultiscaleMapping.hpp"
+#include "mapping/Mapping.hpp"
+#include "math/constants.hpp"
+#include "mesh/Data.hpp"
+#include "mesh/Mesh.hpp"
+#include "mesh/SharedPointer.hpp"
+#include "mesh/Utils.hpp"
+#include "mesh/Vertex.hpp"
+#include "testing/TestContext.hpp"
+#include "testing/Testing.hpp"
+
+using namespace precice;
+using namespace precice::mesh;
+
+BOOST_AUTO_TEST_SUITE(MappingTests)
+BOOST_AUTO_TEST_SUITE(AxialGeoMultiscaleMapping)
+
+BOOST_AUTO_TEST_CASE(testConsistentSpread)
+{
+  PRECICE_TEST(1_rank);
+  int dimensions = 3;
+  using testing::equals;
+
+  // Create mesh to map from
+  PtrMesh inMesh(new Mesh("InMesh", dimensions, testing::nextMeshID()));
+  PtrData inData    = inMesh->createData("InData", 3, 0_dataID);
+  int     inDataID  = inData->getID();
+  Vertex &inVertex0 = inMesh->createVertex(Eigen::Vector3d::Constant(0.0));
+  inMesh->allocateDataValues();
+  Eigen::VectorXd &inValues = inData->values();
+  inValues << 0.0, 0.0, 2.0;
+
+  double radius = 1.0;
+
+  // Create mesh to map to
+  PtrMesh outMesh(new Mesh("OutMesh", dimensions, testing::nextMeshID()));
+  PtrData outData    = outMesh->createData("OutData", 3, 2_dataID);
+  int     outDataID  = outData->getID();
+  Vertex &outVertex0 = outMesh->createVertex(Eigen::Vector3d::Constant(0.0)); // center, equal to incoming mesh node
+  Vertex &outVertex1 = outMesh->createVertex(Eigen::Vector3d(1.0, 0.0, 0.0)); // distance of 1.0 = r to center
+  Vertex &outVertex2 = outMesh->createVertex(Eigen::Vector3d(0.0, 0.5, 0.0)); // distance of 0.5 = r/2 to center
+  outMesh->allocateDataValues();
+
+  // Setup mapping with mapping coordinates and geometry used
+  precice::mapping::AxialGeoMultiscaleMapping mapping(mapping::Mapping::CONSISTENT, dimensions, mapping::AxialGeoMultiscaleMapping::SPREAD, radius);
+  mapping.setMeshes(inMesh, outMesh);
+  BOOST_TEST(mapping.hasComputedMapping() == false);
+
+  // Map data
+  mapping.computeMapping();
+  mapping.map(inDataID, outDataID);
+  const Eigen::VectorXd &outValues = outData->values();
+
+  // Check if data is doubled at center node
```

Comment:
Before the checking, it would be nice to have a comment earlier on regarding the general approach of this test: "Create two meshes so that (blah), then map from `inMesh` to `outMesh` and check if (blah)".

The same for the next test case.

Reply:
Will do!